app/web/src/components/AssetDetailIntrinsicInput.vue

```diff
@@ -157,13 +125,53 @@ const unsetFuncId = computed(() => {
   return func?.funcId as FuncId;
 });
 
+const initialFilter = (): DropdownFilter | null => {
+  if (display.value?.value?.startsWith("s_")) {
+    return "inputSocket";
+  } else if (display.value?.value?.startsWith("p_")) {
+    return "prop";
+  } else if (display.value?.funcId === unsetFuncId.value) {
+    return "unset";
+  } else if ("socketName" in props.data) {
+    // NOTE(nick): fallback to the input data if the display is empty. We need this in case a "emit" blows the
+    // component away. This could be cleaner to avoid having multiple branches.
+    return "inputSocket";
+  } else if ("path" in props.data) {
+    // NOTE(nick): fallback to the input data if the display is empty. We need this in case a "emit" blows the
+    // component away. This could be cleaner to avoid having multiple branches.
+    return "prop";
+  }
```

Comment:
So... the initial filter will be incorrect if an "emit" changed the filter from Unset to either Input Socket or Prop. This happens because it is based on the _displayed_ value, which will be "not set" in both the case of an Input Socket or a Prop. Therefore, we need to fallback and check the incoming data. This could be cleaner, but it works.